File: kosql-execute-async-java/src/main/java/com/github/jacokoo/kosql/execute/async/java/compose/Compose.java
package com.github.jacokoo.kosql.execute.async.java.compose;

import com.github.jacokoo.kosql.compose.expression.Exp;
import com.github.jacokoo.kosql.compose.expression.Functions;
import com.github.jacokoo.kosql.compose.expression.Operator;
import com.github.jacokoo.kosql.compose.typesafe.Selects;

public interface Compose extends Operator, Selects, Insert, Update, Delete, Functions {
    Exp<?> TRUE = Exp.Companion.getTRUE();
}

File: kosql-compose/src/main/kotlin/com/github/jacokoo/kosql/compose/compose.kt
package com.github.jacokoo.kosql.compose

import com.github.jacokoo.kosql.compose.expression.*
import com.github.jacokoo.kosql.compose.statement.Delete
import com.github.jacokoo.kosql.compose.statement.Insert
import com.github.jacokoo.kosql.compose.statement.Update
import com.github.jacokoo.kosql.compose.typesafe.Inserts
import com.github.jacokoo.kosql.compose.typesafe.Selects
import java.math.BigDecimal

abstract class Compose: Operator, Selects, Inserts, Update, Delete, Functions {
    val TRUE = Exp.TRUE
    val INSERT = Insert.INSERT
    val DELETE = Delete.DELETE
}
